lib/kube/proxy/sess_test.go

```diff
@@ -73,99 +72,140 @@ func TestSessionEndError(t *testing.T) {
 	require.NoError(t, err)
 	t.Cleanup(func() { kubeMock.Close() })
 
-	// creates a Kubernetes service with a configured cluster pointing to mock api server
-	testCtx := SetupTestContext(
-		context.Background(),
-		t,
-		TestConfig{
-			Clusters: []KubeClusterConfig{{Name: kubeCluster, APIEndpoint: kubeMock.URL}},
-			// collect all audit events
-			OnEvent: func(event apievents.AuditEvent) {
-				eventsResultMutex.Lock()
-				defer eventsResultMutex.Unlock()
-				eventsResult = append(eventsResult, event)
-			},
+	tests := []struct {
+		name         string
+		recordingErr error
+		interactive  bool
+	}{
+		{
+			name:        "interactive without recording error",
+			interactive: true,
 		},
-	)
+		{
+			name:        "non-interactive without recording error",
+			interactive: false,
+		},
+		{
+			name:         "interactive with recording error",
+			interactive:  true,
+			recordingErr: recordingErr,
+		},
```

Comment:
Should we test non-interactive with recording error?

Reply:
non interactive sessions aren't recorded